From 355107458d4b27117fac93c55534806aeb716545 Mon Sep 17 00:00:00 2001
From: Shinya Kuribayashi <shinya.kuribayashi@necel.com>
Date: Tue, 28 Apr 2009 14:18:37 +0900
Subject: [PATCH] net: nec_candy: Misc cleanups

* Review include headers, remove unneeded inclusions
* Put debug hooks explicitly
* rx_copybreak: Put off by one for consistency with an original driver
* kfree: Give (const void *) cast to get some GCC to shut up

Signed-off-by: Shinya Kuribayashi <shinya.kuribayashi@necel.com>
---
 drivers/net/nec_candy.c |   19 ++++++++-----------
 1 files changed, 8 insertions(+), 11 deletions(-)

diff --git a/drivers/net/nec_candy.c b/drivers/net/nec_candy.c
index 1543f77..ff35a69 100644
--- a/drivers/net/nec_candy.c
+++ b/drivers/net/nec_candy.c
@@ -40,25 +40,18 @@
 #include <linux/types.h>
 #include <linux/kernel.h>
 #include <linux/module.h>
-#include <linux/delay.h>
+#include <linux/interrupt.h>
 #include <linux/netdevice.h>
-#include <linux/param.h>
 #include <linux/etherdevice.h>
-#include <linux/if.h>
 #include <linux/ethtool.h>
 #include <linux/mii.h>
 #include <linux/crc32.h>
-#include <linux/spinlock.h>
-#include <linux/dma-mapping.h>
-#include <linux/timer.h>
 #include <linux/platform_device.h>
 #include <linux/nec_candy_pd.h>
 
 #include <asm/addrspace.h>
-#include <asm/irq.h>
 #include <asm/io.h>
 #include <asm/uaccess.h>
-#include <asm/pgtable.h>
 
 #include "nec_candy.h"
 
@@ -71,6 +64,10 @@
 
 static int debug = -1;
 
+#undef	DEBUG_NEC_CANDY
+#undef	DEBUG_VERBOSE_NEC_CANDY
+#undef	SHOW_BUG		/* targeted bug reporting msgs */
+
 #ifdef DEBUG_VERBOSE_NEC_CANDY
 #define DEBUG_NEC_CANDY
 #endif
@@ -1128,7 +1125,7 @@ candy_rx(struct net_device *dev, ulong isr)
 		newskb->dev = dev;
 		skb_reserve(newskb, 2); /* align IP pkt to 16-byte */
 
-		if (pkt_len <= rx_copybreak) {
+		if (pkt_len < rx_copybreak) {
 			/* we will copy */
 			rxskb = pp->rx_skb[pp->rx_head];
 
@@ -1970,8 +1967,8 @@ static int nec_candy_remove(struct device *dev)
 	for (i = 0; i < RX_RING_SIZE; i++)
 		dev_kfree_skb(pp->rx_skb[i]);
 
-	kfree((void *)KSEG0ADDR(pp->rx_ring));
-	kfree((void *)KSEG0ADDR(pp->tx_ring));
+	kfree((const void *)KSEG0ADDR(pp->rx_ring));
+	kfree((const void *)KSEG0ADDR(pp->tx_ring));
 
 	iounmap((void *)ndev->base_addr);
 	release_mem_region(res->start, res->end - res->start + 1);
-- 
1.6.0.4

